src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -0,0 +1,257 @@
+package com.nutomic.syncthingandroid.preferences;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.net.wifi.WifiConfiguration;
+import android.net.wifi.WifiManager;
+import android.os.Bundle;
+import android.preference.DialogPreference;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.nutomic.syncthingandroid.R;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Dialog preference which allows the user to select on which WiFi networks (based on SSID)
+ * syncing should be allowed.
+ *
+ * Setting can be "All networks" (none selected), or selecting individual networks.
+ *
+ * Due to restrictions in Android, it is possible/likely, that the list of saved WiFi networks
+ * cannot be retrieved if the WiFi is turned off. In this case, an explanation is shown.
+ *
+ * The preference is stored as Set&lt;String&gt; where <code>null</code> represents
+ * "all networks allowed", and a set represents a restriction to the selected SSIDs.
+ *
+ * SSIDs are formatted according to the naming convention of WifiManager, i.e. they have the
+ * surrounding double-quotes (") for UTF-8 names, or they are hex strings (if not quoted).
+ */
+public class WifiSsidPreference extends DialogPreference {
```

Comment:
That it's not possible to dynamically fill the content of the `MultiSelectListPreference` so that it is shown on _current_ invocation. It will only always show the _next_ time you open the preference, because the dialog's `ListView` and adapter are created before the `onPreferenceClick()` is called.

Regarding the `// if`, sure, they slipped from coding-style we use at work, didn't notice I still had so many left in the code.